src/stores/datStore.ts

```diff
@@ -407,19 +410,48 @@ export default class DatStore {
         numToGet: number
     ): Promise<TradeEvent[]> {
         const { library } = this.rootStore.providerStore.getActiveWeb3React();
+        let toBlock = await library.eth.getBlockNumber();
         let fromBlock = (this.rootStore.configStore.getDATinfo()).fromBlock;
-        if (fromBlock > TRADES_FROM_LAST_BLOCKS)
-          fromBlock = (await library.eth.getBlockNumber()) - TRADES_FROM_LAST_BLOCKS;
-          
-        const buyEvents = await this.fetchBuyEvents(datAddress, numToGet, fromBlock);
-        const sellEvents = await this.fetchSellEvents(datAddress, numToGet, fromBlock);
-
-        let combinedTrades: any[] = buyEvents.concat(sellEvents);
-        combinedTrades = combinedTrades.sort(function (a, b) {
-            return b.blockNumber - a.blockNumber;
+        let blockLimit = (this.rootStore.configStore.getDATinfo()).fromBlock;
+        let tradesToReturn = [];
+        
+        if (toBlock > BLOCK_PER_TRADES_FETCH) {
+          blockLimit = toBlock - TRADES_FROM_LAST_BLOCKS;
+          fromBlock = toBlock - BLOCK_PER_TRADES_FETCH;
+        }
+        
+        const self = this;
+        async function getEventsBetweenBlocks() {
+          const buyEvents = await self.fetchBuyEvents(datAddress, numToGet, fromBlock, toBlock);
+          const sellEvents = await self.fetchSellEvents(datAddress, numToGet, fromBlock, toBlock);
+          let combinedTrades: any[] = buyEvents.concat(sellEvents);
+        
+          tradesToReturn = tradesToReturn.concat(combinedTrades);
+
+          console.debug('Getting events between blocks', fromBlock, toBlock, tradesToReturn.length);
+
+          if ((tradesToReturn.length < numToGet) && (fromBlock > blockLimit))
+            if ((fromBlock - BLOCK_PER_TRADES_FETCH) < blockLimit) {
+              fromBlock = blockLimit;
+              toBlock = fromBlock;
+              await getEventsBetweenBlocks();
+            } else {
+              fromBlock = fromBlock - BLOCK_PER_TRADES_FETCH;
+              toBlock = toBlock - BLOCK_PER_TRADES_FETCH;
```

Comment:
`BLOCK_PER_TRADES_FETCH` needs to be updated to `BLOCKS_PER_TRADES_FETCH` in a few places